### PR TITLE
Refactor Security API

### PR DESCRIPTION
**Pull request by 0xh3rman**
Providers:

- [x] Gem
- [x] HashDit
- [x] GoPlus

POST `/v1/scan/security`

Address request:
```json
{
  "target": {
    "address": "0x101ce0cedd142f199c9ef61739ae59b6611a0fc0",
    "chain": "ethereum"
  },
  "type": "address"
}
```

Address Response: 
```json
{
  "malicious": true,
  "reason": "Risk level: 4",
  "provider": "HashDit",
  "metadata": null
}
```

URL request

```json
{
  "target": "https://ftx.us.com"
  "type": "url"
}
```

URL Response
```json
{
  "malicious": true,
  "reason": "Category: phishing_site|Risk level: 5",
  "provider": "GoPlus|HashDit",
  "metadata": null
}
```

### CODE REVIEW

File: apps/api/src/scan_client.rs [hunk 1]
```diff
@@ -20,10 +20,10 @@ impl ScanClient {
             target: ScanTarget::Address(target.clone()),
             metadata: Some(Metadata {
                 name: Some(scan_address.name.unwrap_or_default()),
-                provider: "Gem".to_string(),
                 verified: scan_address.is_verified,
                 required_memo: scan_address.is_memo_required,
             }),
+            provider: "Gem".to_string(),
```
Comment (gemcoder21) [inline apps/api/src/scan_client.rs:26]: let's add it to static/const

Comment (gemcoder21) [reply apps/api/src/scan_client.rs:26]: or list of providers enum

File: crates/security_goplus/src/models.rs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SecurityResponse {
```
Comment (gemcoder21) [inline crates/security_goplus/src/models.rs:5]: should this be generic error response across the app?

**Review by gemcoder21 — APPROVED** [review]

File: crates/security_goplus/src/models.rs (hunk 2 above)
Comment (0xh3rman) [reply crates/security_goplus/src/models.rs:5]: This format is goplus api response, we can define a common one in `primitives` + typeshare for the app

Comment (gemcoder21) [reply crates/security_goplus/src/models.rs:5]: yeah let's use this as as standard for api response

Comment (0xh3rman) [reply crates/security_goplus/src/models.rs:5]: done, added `SecurityResponse` to `primitives` and api returns same model

**Review by gemcoder21 — APPROVED** [review]

File: apps/api/src/scan_client.rs [hunk 3]
```diff
@@ -39,19 +42,20 @@ impl ScanClient {
             }
         }
 
-        // Iterate over security providers
-        for provider in self.security_providers.iter() {
-            let result = provider.scan(&scan_request.target).await;
-            match result {
+        let scanned = future::join_all(self.security_providers.iter().map(|provider| provider.scan(&scan_request.target)))
```
[the remark this answers is not in the document]
Comment (0xh3rman) [reply apps/api/src/scan_client.rs]: added 1s as default timeout, so far the response time is less than 800ms, should be good enough

**Pull request merged**